src/utils/utils.ts

```diff
@@ -457,7 +457,171 @@ export function notInfinity(value: any) {
+ * - Array methods (slice, concat, flat, ...)
+ * - Mathematical operations (+ - % ^ x /)
+ * - Comparison operators (== === != !== < <= > >=)
+ * - The following chars ([ ] ( ) , ; : ? ! .)
+ * - text between simple quotes 'your text'. Only these special chars are allowed : ,:-_;%!?
+ * @param processToExecute A javascript code to execute.
+ * @param variableName variable to use
+ * @returns boolean (true if valid)
+ */
+export function processPassesAllowList(processToExecute: string, variableName: string): boolean {
+    const classes = [
+        'Date',
+        'Math',
+        'Number'
+    ];
+    const dateMethods = [
+        'getDate',
+        'getDay',
+        'getFullYear',
+        'getHours',
+        'getMilliseconds',
+        'getMinutes',
+        'getMonth',
+        'getSeconds',
+        'getTime',
+        'getTimezoneOffset',
+        'getUTCDate',
+        'getUTCDay',
+        'getUTCFullYear',
+        'getUTCHours',
+        'getUTCMilliseconds',
+        'getUTCMinutes',
+        'getUTCSeconds',
+        'toDateString',
+        'toGMTString',
+        'toISOString',
+        'toJSON',
+        'toLocaleDateString',
+        'toLocaleString',
+        'toLocaleTimeString',
+        'toMetaDataString',
+        'toString',
+        'toTimeString',
+        'toUTCString',
+        'valueOf'
+    ];
+
+    const arrayMethod = [
+        'concat',
+        'filter',
+        'find',
+        'findIndex',
+        'flat',
+        'flatMap',
+        'join',
+        'map',
+        'keys',
+        'includes',
+        'reduce',
+        'slice',
+        'toLocaleString',
+        'values'
+    ];
+
+    const stringMethods = [
+        'concat',
+        'endsWith',
+        'includes',
+        'replace',
+        'replaceAll',
+        'slice',
+        'split',
+        'startsWith',
+        'substring',
+        'toLocaleLowerCase',
+        'toLocaleUpperCase',
+        'toLowerCase',
+        'toString',
+        'toUpperCase',
+        'trim',
+        ''
+    ];
+
+    const mathMethods = [
+        'abs',
+        'acos',
+        'asin',
+        'atan',
+        'atan2',
+        'ceil',
+        'cos',
+        'exp',
+        'floor',
+        'log',
+        'max',
+        'min',
+        'pow',
+        'random',
+        'round',
+        'sin',
+        'sqrt',
+        'tan',
+        'trunc',
+        'Mat'
+    ];
+
+    const numberMethods = [
+        'parseFloat',
+        'parseInt',
+        'toExponential',
+        'toFixed',
+        'toPrecision',
+        'isNaN',
+        'isFinite',
+        'isInteger'
+    ];
+
+    const mathOps = [
+        '+', '\\-', '\\/', '*', '%', '^'
+    ];
+
+    const chars = [
+        '\\[', '\\]', '.', ';', ',', '(', ')', ':', '?', '!'
+    ];
+
+    const comparisonOps = [
+        '==', '===', '!=', '!==', '<', '<=', '>', '>='
+    ];
+    const regexPattern: string = `^([ \\t]*(new[ \\t]+(?:${classes.join('|')})` + // use of new + classes
```

Comment:
If i understand well the comments of this pattern, double quotes are not allowed, or is it that we don't have them by design ?

Reply:
there a re allowed too, i add it